src/controllers/ConfirmationController.ts

```diff
@@ -18,10 +18,10 @@ export class ConfirmationController extends BaseAsyncHttpController {
 
         const companyNumber = req.session
             .chain(_ => _.getExtraData())
-            .chainNullable(data => data[AppealKeys.APPEALS_KEY])
-            .chainNullable(appeals => appeals[AppealKeys.PENALTY_IDENTIFIER])
-            .map(penaltyIdentifier => penaltyIdentifier[PenaltyIdentifierKeys.COMPANY_NUMBER])
-            .extract();
+            .chainNullable(data => data.appeals as Appeal)
+            .chainNullable(appeals => appeals.penaltyIdentifier as PenaltyIdentifier)
```

Comment:
Why do you need to cast it as `PenaltyIdentifier`?

Reply:
If TS is not able to interfere type it would be better to do `(appeal: Appeal) => ...`.

By the way I used singular form because there is one appeal only.